src/api_manager/auth/authz_cache.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) Extensible Service Proxy Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+#include "src/api_manager/auth/authz_cache.h"
+
+namespace google {
+namespace api_manager {
+namespace auth {
+
+namespace {
+// The maximum lifetime of a cache entry. Unit: seconds.
+const int kAuthzCacheTimeout = 300;
+// The number of entries in authz cache.
+const int kAuthzCacheSize = 100;
+}  // namespace
+
+AuthzCache::AuthzCache() : cache_(kAuthzCacheSize) {}
+
+AuthzCache::~AuthzCache() { cache_.Clear(); }
+
+void AuthzCache::Add(const std::string& cache_key, const bool if_success,
+                     const std::chrono::system_clock::time_point& token_exp,
```

Comment:
I thought we are NOT using token expiration.  we always use 5 minutes expiration